src/test/java/it/gov/pagopa/reward/notification/event/consumer/IbanOutcomeRecoveryIntegrationTest.java

```diff
@@ -0,0 +1,259 @@
+                ));
+
+
+        notificationRuleList = notificationRuleRepository.saveAll(
+                        IntStream.rangeClosed(0, 15).mapToObj(i -> {
+                            RewardNotificationRule x = RewardNotificationRuleFaker.mockInstance(i);
+                            x.setInitiativeId(INITIATIVEID.formatted(i));
+
+                            switch (i % 4) {
+                                case 0 -> // expired
+                                        x.setEndDate(today.minusDays(5));
+                                case 1 -> { // timeParameter
+                                    x.setEndDate(today.plusDays(5));
+                                    x.setAccumulatedAmount(null);
+                                }
+                                case 2 -> { // budgetExhausted
+                                    x.setEndDate(today.plusDays(5));
+                                    x.setTimeParameter(null);
+                                    x.setAccumulatedAmount(AccumulatedAmountDTO.builder()
+                                            .accumulatedType(AccumulatedAmountDTO.AccumulatedTypeEnum.BUDGET_EXHAUSTED).build());
+                                }
+                                case 3 -> { // threshold
+                                    x.setEndDate(today.plusDays(5));
+                                    x.setTimeParameter(null);
+                                }
+                            }
+
+                            return x;
+                        }).toList())
+                .collectList().block();
+    }
+
+    @AfterEach
+    void cleanUp() {
+        for (IbanOutcomeDTO iban : ibanOutcomeDTOList) {
+            rewardIbanService.deleteIban(iban).block();
+        }
+
+        rewardsNotificationRepository.deleteAll(rewardsNotificationList).block();
+        for (RewardsNotification r : rewardsNotificationList) {
+            rewardsNotificationRepository.deleteById(getRemedialNotificationId(r.getId())).block();
+        }
+
+        notificationRuleRepository.deleteAll(notificationRuleList).block();
+    }
+
+    @Test
+    void test() {
+        // notificationId = USERID%s_INITIATIVEID%s_today.format(Utils.FORMATTER_DATE)
+
+        ibanOutcomeDTOList.forEach(p -> publishIntoEmbeddedKafka(topicIbanOutcome, null, p.getUserId().concat(p.getInitiativeId()), p));
+
+        waitForIbanStoreChanged(totalIban, rewardIbanRepository);
+
+        for (RewardsNotification r : rewardsNotificationList) {
+            RewardsNotification recovered = rewardsNotificationRepository.findById(r.getId()).block();
+            Assertions.assertNotNull(recovered);
+
+            int i = getIndexFromNotificationId(recovered.getId());
+            if (i >= 0 && i <= 3) { // status ERROR rejectionReason IBAN_NOT_FOUND - [0,3]
+                checkRecoveredErrorStatus(recovered);
+                checkNotificationDate(i, recovered.getNotificationDate(), recovered.getId());
```

Comment:
as the checkNotificationDate has been put inside the checkRecoveredCompletedKo, we could move its call also inside the checkRecoveredErrorStatus